### DISCUSSION TITLE
Help setting up GPU hardware acceleration.

### DISCUSSION THREAD

**Discussion by mechpanos** [blakeblackshear/frigate#1068] [Ask A Question]
Hi, I would like some help here!
I am running the addon, and can't setup gpu hardware acceleration, to offload my cpu.
I have an ATI 5450, I set up 
```
ffmpeg:
  hwaccel_args:
    - -hwaccel
    - vaapi
    - -hwaccel_device
    - /dev/dri/renderD128
```
 But then, all I get is green video.
I try to set the variable in docker (through Portainer), to **_LIBVA_DRIVER_NAME=radeonsi_**
but no success! Then I check the container, and it is set back to "I965", by itself. Is Supervisor, making this?
What else, should I try?
I have some experience with Linux, but none with Docker (that's why I installed Portainer, to help me get along with a GUI to manage images and containers).

**Comment by MattEng0** [comment]
Cause you are running it through HomeAssistant in order to set the environment variables you have to add them to the config.yml file so it isn't overwritten by the defaults on start up,

So in your case add this line:
 ```yaml 
environment_vars:
   LIBVA_DRIVER_NAME=radeonsi
```

More details listed here [Frigate: Advanced Configuration](https://blakeblackshear.github.io/frigate/configuration/advanced)

**Comment by DanGarion** [comment]
I have a question about _where_ the environment variable goes. I'm running the Frigate add-on within HA and when I add 

```
environment_vars:
  LIBVA_DRIVER_NAME=radeonsi
```
To my frigate.yml I get errors saying you can't put that in there. Am I supposed to put it in the frigate.yml or in my HA configuration.yml? I tried HA config and that didn't check out either.

Here is my error.

`[2022-06-30 08:58:24] frigate.app                    INFO    : Starting Frigate (0.11.0-1d45b0b)
*************************************************************
*************************************************************
***    Your config file is not valid!                     ***
***    Please check the docs at                           ***
***    https://docs.frigate.video/configuration/index     ***
*************************************************************
*************************************************************
***    Config Validation Errors                           ***
*************************************************************
1 validation error for FrigateConfig
environment_vars
  value is not a valid dict (type=type_error.dict)
Traceback (most recent call last):
  File "/opt/frigate/frigate/app.py", line 326, in start
    self.init_config()
  File "/opt/frigate/frigate/app.py", line 82, in init_config
    user_config = FrigateConfig.parse_file(config_file)
  File "/opt/frigate/frigate/config.py", line 942, in parse_file
    return cls.parse_obj(config)
  File "pydantic/main.py", line 521, in pydantic.main.BaseModel.parse_obj
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for FrigateConfig
environment_vars
  value is not a valid dict (type=type_error.dict)
*************************************************************
***    End Config Validation Errors                       ***
*************************************************************`

**Reply by NickM-27** [reply]
That's because the format is wrong, it needs to be a dict:

```yaml
environment_vars:
  LIBVA_DRIVER_NAME: radeonsi
```

**Reply by DanGarion** [reply]
Oh man... I totally missed that. Thank you. Not sure if that improved anything on the CPU side though.
I have a AMD GX-415GA SOC with Radeon(tm) HD Graphics, 4 cores and it seems to peg it at 122% CPU regardless... 👯

**Reply by NickM-27** [reply]
Try using radeontop to ensure the GPU is being used

**Reply by DanGarion** [reply]
Ok so instead of radeonsi change it to radeontop. I will try that!

**Reply by NickM-27** [reply]
No no sorry I wasn't clear, `radeontop` is a program you can run from the command line that will show your GPU and different processes. So you can tell what % is being used (and if it is 0, then frigate is not utilizing the GPU)

**Reply by DanGarion** [reply]
Hah... ok :)

**Reply by DanGarion** [reply]
So. I don't think it is working through Frigate. 
![image](https://user-images.githubusercontent.com/1666375/176735317-b1c92e24-f372-4adb-8d3d-194861d41450.png)

**Reply by NickM-27** [reply]
Looks like it isn't, this is what I see

![Screen Shot 2022-06-30 at 11 01 08 AM](https://user-images.githubusercontent.com/14866235/176735576-a213fe19-4f3f-4487-bcd5-885e3d81d47b.png)

**Reply by DanGarion** [reply]
Well let me look a little deeper in on my CPU, maybe it doesn't use the setup I assumed it did. I do see the render128 driver in dev/dri so I guess I will either just deal with it or figure it out. :)

**Reply by NickM-27** [reply]
You'll also want to make sure that the host OS contains the driver for the iGPU. For example, I run Unraid and had to include it so frigate could utilize the iGPU.

**Reply by DanGarion** [reply]
I'm runnnig debian 10. I checked to see if I have drivers loaded and it seems like I do. I will keep digging. I'm not "seeing" any issues other than it seems like it isn't using the GPU. I do have a coral/tpu as well and I know that is working.

**Reply by DanGarion** [reply]
Oh I think I know why radeontop won't show my GPU usage it only works with the free drivers (for anything more than the mem), I have official drivers installed. I'll just assume it is working until I see an issue.

**Reply by NickM-27** [reply]
Ah very interesting
